test/01_contraction/contraction_test_helpers.hpp

```diff
@@ -68,9 +68,119 @@ auto inline load_config_params()
     return testParams;
 }
 
+bool checkMemoryLimit(uint64_t minMemory,
+                      uint64_t maxMemory,
+                      uint64_t rank,
+                      uint64_t i = 1,
+                      uint64_t j = 1,
+                      uint64_t k = 1,
+                      uint64_t x = 1,
+                      uint64_t y = 1,
+                      uint64_t z = 1)
+{
+    uint64_t size = 4 * pow(i * j * k * x * y * z, 2);
+
+    return (size >= minMemory) && (size <= maxMemory);
+}
+
+void generate_lengths(std::vector<std::vector<size_t>>               ranges,
+                      std::vector<std::vector<std::vector<size_t>>>& lengths,
+                      uint32_t                                       rank)
+{
+    // Minimum and maximum memory footprint of each tensor
+    uint64_t minMemory = 64;
```

Comment:
These are not quite indicative of memory - I would interpret them more of an element count limit.
Probably best to have them as static const as well